src/controllers/help.controller.ts

```diff
@@ -0,0 +1,66 @@
+import { Router, Request, Response } from 'express';
+import { Controller } from '../interfaces';
+import axios from 'axios';
+import MarkdownIt from 'markdown-it';
+
+const md = new MarkdownIt();
+
+const fetchCommands = async (user, repo) => {
+    try {
+        const url = `https://api.github.com/repos/${user}/${repo}/contents/commands.md`;
+        const response = await axios.get(url, {
+            headers: {
+                'Accept': 'application/vnd.github.v3.raw',
+            },
+        });
+        return response.data;
+    } catch (error) {
+        console.error(`Error fetching commands: ${error}`);
```

Comment:
@princerajpoot20 we should handle more specific errors from the GitHub API, how do you think we could improve error handling here?

Reply:
I have checked different error types in the Github API:

## Client Errors (4xx)

- 400 Bad Request: The request was invalid or cannot be served. 
- 401 Unauthorized: The user does not have the necessary credentials. 
- 403 Forbidden: The user is authenticated but does not have the permissions to perform the requested action.
- 404 Not Found: The requested resource could not be found.
- 422 Unprocessable Entity: The request was well-formed, but was unable to be followed due to semantic errors. For example, this could be due to validation errors in the data provided.
- 429 Too Many Requests: You have exceeded the rate limits.

## Server Errors (5xx)

- 500 Internal Server Error: An unexpected error occurred on GitHub's servers.
- 502 Bad Gateway, 503 Service Unavailable, 504 Gateway Timeout: These indicate that there is a problem with GitHub's servers.

I will handle each error separately so that we can improve error handling.

Reply:
@princerajpoot20 200 status code is not an error 🙂, 3xx isn't either. If you get a 3xx you probably won't have an exception raised, but it's a valid test for you to do. For example if the `usage.md` file is deleted or moved 👍, it would be good to handle that case.

You can test this 3xx scenario with your own repo like you're currently doing. 

About 5xx errors, I'd like to challenge you to specify in more detail how you could improve error handling 😃? Think about [software quality attributes](https://github.com/joelparkerhenderson/system-quality-attributes).

Reply:
@princerajpoot20 I noticed now that you changed this function to get the openapi.yaml, since we're using it to get commands. We can use the `getAppOpenAPI` util function on this repository (it's already used elsewhere please take a look), that way you don't make an HTTP request and there is no rate limiting or other logic required 👍 , wdyt about refactoring to using that function @princerajpoot20 ?